SAKPaaS/src/app/core/services/gps.service.ts

```diff
@@ -1,34 +1,122 @@
 
-  private coordinates = new Subject<GpsCoordinates>();
+  private coordinates = new BehaviorSubject<GpsCoordinates>(GpsService.HOME_LOCATION);
 
-  constructor() { }
+  constructor(
+    private snackBarService: SnackBarService,
+    private cookieService: CookieService,
+    private dialogService: GlobalDialogService
+  ) {
+    if (!this.isCookieAlreadySet()) {
+      this.askForPermission().then((granted) => {
+        if (granted) {
+          this.cookieService.set(GpsService.PERMISSION_COOKIE_NAME, 'true', 5);
+          this.updateRealGpsPostion();
+        } else {
+          console.warn('Access to GPS position denied. What to do?');
+        }
+      });
+    } else {
+      this.updateRealGpsPostion();
+    }
+  }
 
   public setLocation(coordinates: GpsCoordinates) {
     coordinates.fromDevice = false;
     this.coordinates.next(coordinates);
   }
 
   public getLocation(): Observable<GpsCoordinates> {
+    return this.coordinates;
+  }
+
+  public getCurrentLocation(): GpsCoordinates {
+    return this.coordinates.value;
+  }
+
+  private updateRealGpsPostion() {
     if (navigator.geolocation) {
       navigator.geolocation.getCurrentPosition((position) => {
         const longitude = position.coords.longitude;
         const latitude = position.coords.latitude;
         console.log(longitude + ' - ' + latitude);
         this.coordinates.next({ longitude, latitude, fromDevice: true });
+      }, (positionError) => {
+        switch (positionError.code) {
+          case positionError.PERMISSION_DENIED:
+            this.snackBarService.sendNotification({
+              message: 'Bitte erlaube HappyHamster, deine Position zu ermitteln.\
+                  Lade dazu entweder die Seite neu oder gehe in die Browser-Einstellungen.',
+              type: SnackBarTypes.ERROR
+            });
+            break;
+          case positionError.POSITION_UNAVAILABLE:
+            this.snackBarService.sendNotification({
+              message: 'HappyHamster konnte deine GPS-Position nicht ermitteln.',
+              type: SnackBarTypes.ERROR
+            });
+            break;
+          case positionError.TIMEOUT:
+            this.snackBarService.sendNotification({
+              message: 'HappyHamster konnte deine GPS-Position leider nicht ermitteln. Versuche es erneut.',
+              type: SnackBarTypes.ERROR
+            });
+            break;
+          default:
+            this.snackBarService.sendNotification({
+              message: 'HappyHamster konnte aus einem unbekannten Grund deine GPS-Position nicht ermitteln.',
+              type: SnackBarTypes.ERROR
+            });
+            break;
+        }
       });
-      return this.coordinates;
     } else {
+      this.snackBarService.sendNotification(
+        { message: 'Wir konnten deine GPS-Koordinaten nicht abrufen, da dein Browser das nicht unterstützt :(', type: SnackBarTypes.ERROR }
```

Comment:
Zeilenumbruch

Reply:
Sonst ist die Zeile aber zu lang :(

Reply:
Hä, welche Zeile ist zu lang, wenn du type in eine neue Zeile bringst? :D
```
this.snackBarService.sendNotification({
    message: 'Wir konnten deine GPS-Koordinaten nicht abrufen, da dein Browser das nicht unterstützt :(',
    type: SnackBarTypes.ERROR
});
```

Reply:
Ja war vorher alles ein Einzeiler xD Hab's gefixt